### PR TITLE
(#4) Display Best Bets

### LINKED ISSUE #4: Story: Display Best Bets
## As a content editor, I would like to identify content for users to find that are related to their search term, in order to aid them in their search for cancer information.

> **ESTIMATE** 3

### Acceptance Criteria

```gherkin
Scenario: English Best Bets Display 
  Given user navigates to cancer.gov
  Given user navigates to "/?swKeyword=CTEP"
  Then the system returns the results page for CTEP
    Then the H1 appears as “NCI Search Results”
    And “Results for: CTEP” appears below the H1
    And a blue box for Best Bets appears below the “Results for: CTEP” text
    And the text inside the the box appears as an H2 as “Best Bets for Cancer Therapy Evaluation Program (CTEP)”
      Where “Cancer Therapy Evaluation Program (CTEP)” is the text that matched the search term from the API
    And the title of the related item appears as a link below the H2 from the API
    And the description of the item appears below the title from the API
    Then “Results 1–10 of <total terms returned> for: CTEP” appears as an h4
    Then the results, bottom pager and number of results display option are displayed (from previous stories)
  When user navigates to page 2 or more of the search results 
    Then best bets no longer appear at the top of the page (they only appear on the first page of results)
Examples:
|search term| |type|
|Clinical trials ideas| |single best bet, single link|
|cancer risk assessment tools| |single best bet topic, multiple links|
```

```gherkin
Scenario: Spanish Best Bets Display 
  Given language is set to “es”
  Given user navigates to "/?swKeyword=Nacional de Estudios Clínicos"
  Then the system returns the results page for Nacional de Estudios Clínicos
  Then the H1 appears as “Resultados”
    And “Resultados para: Nacional de Estudios Clínicos” appears as an H4 below the H1
    And a blue box for Best Bets appears below the “Resultados para: Nacional de Estudios Clínicos” text
    And the text inside the the box appears as an H2 as “Mejores resultados para Estudios clínicos”
      Where “Estudios clinicos” is the text that matched the search term from the API
    And the title appears as a link below the H2 from the API
    And the description of the item appears below the title from the API
    Then “Results 1–10 of <total terms returned> for: Nacional de Estudios Clínicos” appears as an h4
    Then the results, bottom pager and number of results display option are displayed (from previous stories)
  When user navigates to page 2 or more of the search results 
    Then best bets no longer appear at the top of the page (they only appear on the first page of results)
Examples
|search term| |type|
|Exploración con TEP| |single link|
|tumor primario desconocido| |single best bet, multiple links|
|Medicamentos para el cancer| |multiple best bet categories, multiple links|
```

**TODO: Add Test for 2 Categories**

```gherkin
Scenario: Best Bets Do Not show on Microsites
  Given the SWS API is <url>
  And Best Bets API is not set
  When the user views search results
    Then no best bets should appear
Examples:
|url|
|dceg.cancer.gov|
| cancer.gov/nano|
| cancer.gov/pediatric-adult-rare-tumor|
| cancer.gov/rare-brain-spine-tumor/|
| cancer.gov/rare-brain-spine-tumor/espanol|
```

```gherkin
Click analytics
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [x] Add `BestBet` component
- [x] Api call for best bet
- [x] Display results returned from api call
- [x] Style `BestBet`
- [x] Unit test
- [ ] Integration tests

### Technical Notes
Screen Shot 2020-07-28 at 11 18 11 AM

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #4

* Added empty `BestBet` component
* Added API call
* Displayed API results
* Styled `BestBet` component
* Added language support
* Added unit test for `BestBet`
* Added support for multiple best bet items
* Added array length check
* Added integration tests, updated proxy setup
* Updated styling for display of Definition and BestBet together
* Updated mock files to conform with Search request, updated variable name casing
* Put back removed styling for BestBet/Definition display, do not make best bet api call if not configured, removed unused vars
* Added pos and neg tests for best best display

### CODE REVIEW

**Comment by seyilonge-nci** [comment]
@VictoriaSunNIH @blilianyu @mworrest
Please review: https://react-app-dev.cancer.gov/sitewide-search-app/pr-32?cfg=0&swkeyword=ctep

**Comment by VictoriaSunNIH** [comment]
@lburack tagging you too for best bets (your favorite!)

**Comment by lburack** [comment]
English and Spanish CGOV Best Bets are appearing as expected.
Best Bets do not appear for NCI-CONNECT (EN and ES) and DCEG, as expected.
Passes UX Review!

File: src/components/molecules/best-bet/best-bet.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.definition + .best-bet {
+	margin-top: 1.5rem;
```
Comment (arcepaul) [inline src/components/molecules/best-bet/best-bet.scss:2]: is this supposed to be rem?

**Pull request merged**